src/components/FileRepo/DownloadButton.tsx

```diff
@@ -0,0 +1,216 @@
+/* eslint-disable react/prop-types */
+import React, { Fragment, FunctionComponent } from 'react';
+// @ts-ignore
+import { compose } from 'recompose';
+// @ts-ignore
+import { injectState } from 'freactal';
+import uniq from 'lodash/uniq';
+// @ts-ignore
+import { ColumnsState } from '@kfarranger/components/dist/DataTable';
+import { familyMemberAndParticipantIds } from '../FamilyManifestModal';
+import { TRACKING_EVENTS, trackUserInteraction } from 'services/analyticsTracking';
+import {
+  clinicalDataFamily,
+  clinicalDataParticipants,
+  downloadBiospecimen,
+} from 'services/downloadData';
+import { withApi } from 'services/api';
+import FamilyManifestModal from '../FamilyManifestModal/FamilyManifestModal';
+
+import './DownloadButton.css';
+import { Menu, message, Spin } from 'antd';
+import ReportsButton from 'ui/ReportsButton';
+import { Sqon, ModalStateEffects } from '../../types';
+
+type Column = {
+  Header: string;
+  accessor: string;
+  canChangeShow: boolean;
+  extendedDisplayValues: any;
+  extendedType: string;
+  field: string;
+  jsonPath: any;
+  query: any;
+  show: boolean;
+  sortable: boolean;
+  type: string;
+  [key: string]: any;
+};
+
+type DownloaderParams = {
+  api?: Function;
+  sqon: Sqon;
+  columns: Column[];
+};
+
+type StateProp = {
+  type: string;
+  keyField: string;
+  defaultSorted: any;
+  columns: Column[];
+  [key: string]: any;
+};
+
+type Props = {
+  sqon: Sqon;
+  api: Function;
+  projectId: string;
+  effects: ModalStateEffects;
+};
+
+type ColumnStateRenderProps = {
+  state: StateProp;
+  loading: boolean;
+};
+
+const trackDownload = async (label: string) =>
+  // @ts-ignore
+  await trackUserInteraction({
```

Comment:
can you provide with more information on why this is not solvable here? `@ts-ignore` I know it's sometime a necessary evil, but we should justify it a little at least.

Reply:
is it easily fixable, need major refactoring?

Reply:
I can not completely remove the 'ts-ignore' but I improved the situation based on your comment. It will be fixed when we refactor the tracking function from plain javascript into typescript.